client/task-list/tasks/products/product-template-modal.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button, Modal, RadioControl } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { applyFilters } from '@wordpress/hooks';
+import { ITEMS_STORE_NAME } from '@woocommerce/data';
+import { getAdminLink } from '@woocommerce/wc-admin-settings';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import './product-template-modal.scss';
+import { createNoticesFromResponse } from '../../../lib/notices';
+
+export const ONBOARDING_PRODUCT_TEMPLATES_FILTER =
+	'woocommerce_admin_onboarding_product_templates';
+
+const PRODUCT_TEMPLATES = [
+	{
+		key: 'physical',
+		title: __( 'Physical product', 'woocommerce-admin' ),
+	},
+	{
+		key: 'digital',
+		title: __( 'Digital product', 'woocommerce-admin' ),
+	},
+	{
+		key: 'variable',
+		title: __( 'Variable product', 'woocommerce-admin' ),
+	},
+];
```

Comment:
Currently left as a hardcoded list here.
Depending on customizability, this could be changed to registered configs on the back-end, that we retrieve with the API. I don't think that is quite necessary atm.